### PR TITLE
Evaluation dashboard charts

### PR DESCRIPTION
**Pull request by csansoon**
image

TODO: Tweak the chart container to better display the loading and error states.

### CODE REVIEW

File: apps/web/package.json [hunk 1]
```diff
@@ -44,6 +45,7 @@
     "react": "19.0.0-rc-f994737d14-20240522",
     "react-dom": "19.0.0-rc-f994737d14-20240522",
     "socket.io-react-hook": "^2.4.5",
+    "recharts": "2.13.0-alpha.5",
```
Comment (geclos) [inline apps/web/package.json:48]: recharts at the end?

File: apps/web/src/actions/evaluationResults/computeAggregatedResults.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const evaluation = await evaluationScope
+      .find(evaluationId)
+      .then((r) => r.unwrap())
+
+    return await computeAverageResultOverTime({
+      workspaceId: workspace.id,
+      evaluation,
+      documentUuid,
+    }).then((r) => r.unwrap())
+  })
```
Comment (geclos) [inline apps/web/src/actions/evaluationResults/computeAggregatedResults.ts:48]: nice 👌🏼

**Review by geclos — APPROVED** [review]
looks coolio!

File: apps/web/package.json (hunk 1 above)
Comment (andresgutgon) [reply apps/web/package.json:48]: Is used under the hood by shad/cn charts

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/CostOverResults.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  documentUuid: string
+}) {
+  const { isLoading, error, data } = useAverageResultsAndCostOverCommit({
+    evaluation,
+    documentUuid,
+  })
+  const { commit } = useCurrentCommit()
+
+  useEffect(() => {
+    console.log({ data, isLoading, error })
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/CostOverResults.tsx:30]: 🔥

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/CostOverResults.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              min: evaluation.configuration.detail!.range.from,
+              max: evaluation.configuration.detail!.range.to,
+            },
+            yAxis: {
+              label: 'Average cost',
+              type: 'number',
+              min: 0,
+              max: 'auto',
+            },
+            data: data!.map((r) => ({
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/CostOverResults.tsx:50]: Maybe useMemo here

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/ResultsOverTime.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+              max: maxDate,
+              tickFormatter: (value) => formatDate(value as number),
+            },
+            yAxis: {
+              label: 'Average result',
+              type: 'number',
+              min: evaluation.configuration.detail!.range.from,
+              max: evaluation.configuration.detail!.range.to,
+            },
+            data: data!.map((r) => ({
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/ResultsOverTime.tsx:64]: Same, useMemo. Just take into account if can help

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/ResultsOverTime.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                  </div>
+                </div>
+              )
+            },
+          }}
+        />
+      ) : (
+        <div className='flex w-full h-full items-center justify-center'>
+          <Text.H6 color='foregroundMuted'>No data</Text.H6>
+        </div>
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/ResultsOverTime.tsx:91]: This is repeated. You can do a generic blank slate

File: packages/web-ui/src/ds/molecules/Charts/AreaChart/index.tsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+      const { width, height } = containerRef.current!.getBoundingClientRect()
+      setDimensions({ width, height })
+    })
+
+    resizeObserver.observe(containerRef.current)
+
+    return () => {
+      resizeObserver.disconnect()
+    }
+  }, [])
```
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/web-ui/src/ds/molecules/Charts/AreaChart/index.tsx:45]: I can DRY lots of things between `AreaChart`, `ScatterChart`, and any more to come, they all follow the exact same structure with few changes. Will do in another PR in the future.

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/CostOverResults.tsx (hunk 4 above)
Comment (csansoon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/MetricsSummary/Charts/Numerical/CostOverResults.tsx]: 🤝

**Pull request merged**